src/components/CreateVault.tsx

```diff
@@ -72,16 +72,20 @@ const useVaultInputValidation = () => {
   }
 
   if (selectedMetrics) {
-    const collateralPurse = (purses ?? []).find(
-      ({ brand }) => brand === selectedMetrics.totalCollateral.brand,
-    );
-
-    if (
-      !collateralPurse ||
-      (collateralPurse.currentAmount as Amount<'nat'>).value <
-        (valueToLock ?? 0n)
-    ) {
-      toLockError = 'Need to obtain funds';
+    if (!purses) {
```

Comment:
I think our style is to avoid unnecessary negations. so,
```js
if (purses) {
  …
} else {
 toLockError = 'Need to connect wallet';
}
```

Reply:
Looking more closely at `useVaultInputValidation` only the first handful of lines need to be in a hook. The rest of this logic would be better extracted into a function that could be tested as a unit. eg..
```js
const useVaultInputValidation = () => {
  return detectInputErrors({
    collateralizationRatio: useAtomValue(collateralizationRatioAtom),
    selectedCollateralId: useAtomValue(selectedCollateralIdAtom),
    valueToReceive: useAtomValue(valueToReceiveAtom),
    valueToLock: useAtomValue(valueToLockAtom),
    purses: useAtomValue(pursesAtom),
    vaultStoreState: useVaultStore.getState(),
  })
};
```

Reply:
I should've just made this a computed atom rather than hook in the first place, so just rewrote it as such. The atoms are pretty testable as well, probably even more so since we wouldn't have to map the args manually in the test.